internal/integrationtest/main/main_test.go

```diff
@@ -0,0 +1,160 @@
+		ArduinoCLIPath:         paths.New("..", "..", "..", "arduino-cli"),
+		UseSharedStagingFolder: true,
+	})
+
+	// Run version and check the output message
+	stdout, stderr, err := cli.Run("version")
+	require.NoError(t, err)
+	require.Contains(t, string(stdout), "Version")
+	require.Contains(t, string(stdout), "Commit")
+	require.Empty(t, stderr)
+
+	// Checks if "version --format json" has a json as an output
+	stdout, _, err = cli.Run("version", "--format", "json")
+	require.NoError(t, err)
+	var jsonMap map[string]string
+	err = json.Unmarshal(stdout, &jsonMap)
+	require.NoError(t, err)
+
+	// Checks if Application's value is arduino-cli
+	require.Equal(t, jsonMap["Application"], "arduino-cli")
+
+	// Checks if VersionString's value is git-snapshot, nightly or a valid semantic versioning
+	switch version := jsonMap["VersionString"]; version {
+	case "git-snapshot":
+		require.Contains(t, version, "git-snapshot")
+	case "nigthly":
+		require.Contains(t, version, "nightly")
+	default:
+		_, err = semver.Parse(version)
+		require.NoError(t, err)
+	}
+
+	// Checks if Commit's value is not empty
+	require.NotEmpty(t, jsonMap["Commit"])
+}
+
+func TestLogOptions(t *testing.T) {
+	// Using version as a test command
+	env := testsuite.NewEnvironment(t)
+	defer env.CleanUp()
+
+	cli := integrationtest.NewArduinoCliWithinEnvironment(env, &integrationtest.ArduinoCLIConfig{
+		ArduinoCLIPath:         paths.New("..", "..", "..", "arduino-cli"),
+		UseSharedStagingFolder: true,
+	})
+
+	// No logs
+	stdout, _, _ := cli.Run("version")
+	trimOut := strings.TrimSpace(string(stdout))
+	outLines := strings.Split(trimOut, "\n")
+	require.Len(t, outLines, 1)
+
+	// Plain text logs on stdout
+	stdout, _, _ = cli.Run("version", "-v")
+	trimOut = strings.TrimSpace(string(stdout))
+	outLines = strings.Split(trimOut, "\n")
+	require.Greater(t, len(outLines), 1)
+	require.True(t, strings.HasPrefix(outLines[0], "\x1b[36mINFO\x1b[0m")) // account for the colors
+
+	// Plain text logs on file
+	logFile := cli.DataDir().Join("log.txt")
+	_, _, _ = cli.Run("version", "--log-file", logFile.String())
+	lines, _ := logFile.ReadFileAsLines()
+	require.True(t, strings.HasPrefix(lines[0], "time=\""))
+
+	// json on stdout
+	stdout, _, _ = cli.Run("version", "-v", "--log-format", "JSON")
+	trimOut = strings.TrimSpace(string(stdout))
+	outLines = strings.Split(trimOut, "\n")
+	requirejson.Contains(t, []byte(outLines[0]), `{ "level" }`)
+
+	// Check if log.json contains readable json in each line
+	var v interface{}
+	logFileJson := cli.DataDir().Join("log.json")
+	_, _, _ = cli.Run("version", "--log-format", "JSON", "--log-file", logFileJson.String())
+	fileContent, err := logFileJson.ReadFileAsLines()
+	require.NoError(t, err)
+	for _, line := range fileContent {
+		// it seems that vscode has problems reading a stream of JSONs. In this case, the last line of the file is empty and
+		// causes the check to fail
```

Comment:
Who cares about vscode?
the problem here is that `json.Unmarshal` considers an empty line invalid json.

```suggestion
		// exclude empty lines since they are not valid json
```